### PR TITLE
chore(workflow-v2.4.0): releasing workflow-v2.4.0(-e2e)

### PR DESCRIPTION
**Pull request by mboersma**
Replaces #335 since CI expects to see this branch in the deis org repo.

### CODE REVIEW

File: workflow-v2.4.0-e2e/tpl/workflow-e2e-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - name: tests
+    image: quay.io/{{.e2e.org}}/workflow-e2e:{{env "WORKFLOW_E2E_GIT_TAG" | default .e2e.dockerTag}}
+    imagePullPolicy: {{.e2e.pullPolicy}}
+    env:
+      - name: GINKGO_NODES
+        value: "{{env "GINKGO_NODES" | default 15}}"
+      - name: JUNIT
+        value: "true"
+      - name: CLI_VERSION
+        value: "{{env "CLI_VERSION" | default "latest"}}"
```
Comment (bacongobbler) [inline workflow-v2.4.0-e2e/tpl/workflow-e2e-pod.yaml]: should this be hardcoded to v2.4.0?

Comment (mboersma) [inline workflow-v2.4.0-e2e/tpl/workflow-e2e-pod.yaml]: Yes, probably. Let me try that locally to make sure it's a safe change.

Comment (bacongobbler) [inline workflow-v2.4.0-e2e/tpl/workflow-e2e-pod.yaml]: FYI I tried manually with `v2.4.0` as the value and it worked for me :)

File: workflow-v2.4.0/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: workflow-dev
```
Comment (vdice) [inline workflow-v2.4.0/Chart.yaml]: `dev` should be `v2.4.0`

File: workflow-v2.4.0/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+home: https://github.com/deis/workflow
+version: dev
```
Comment (vdice) [inline workflow-v2.4.0/Chart.yaml]: `dev` should be `v2.4.0`

File: workflow-v2.4.0-e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Workflow 2.4.0-dev e2e Tests (parallel)
```
Comment (vdice) [inline workflow-v2.4.0-e2e/README.md]: should this `-dev` be removed?

Comment (mboersma) [inline workflow-v2.4.0-e2e/README.md]: Yep--good catch, I'll clean up all that text.

**Pull request merged**